### DISCUSSION TITLE
CalcGrains has changed since 5.4?

### DISCUSSION THREAD

**Discussion by neilmanck** [mtex-toolbox/mtex#799] [Ask Anything] [answered]
Hi Ralf,

It seems that the results of CalcGrains are different after 5.3? I include two snips of parts of a figure for plot(grains) for which the code is identical in each case - only one is with 5.3 and one is with 5.6.

%% construct the grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',10*degree);
%% smooth the grains
grains = smooth(grains,5);
%% plot all the grains as a phase map
figure
plot(grains)
![Snip_5 3](https://user-images.githubusercontent.com/75994459/108600718-040bba00-7399-11eb-9596-047f8feb5a47.JPG)
![Snip_5 6](https://user-images.githubusercontent.com/75994459/108600722-0bcb5e80-7399-11eb-9801-6dbdc56c719e.JPG)

regards,
Neil

**Comment by ralfHielscher** [comment, marked as answer]
The function calcGrains did not change so much. The function smoothed changed a bit. It good options to control whether triple points move or not. While in older MTEX versions triple points could move the default in the current version is that they do not move.

Ralf.

**Reply by neilmanck** [reply]
Thanks Ralf.
Indeed if I now use
grains = smooth(grains,5,'moveTriplePoints');
in MTEX 5.6 I get a result that looks to my eyes identical to what I was getting in MTEX 5.3 with the default settings.
cheers,
Neil
